### ISSUE TITLE
Indicate Obsolete Declension in Inflection Templates

### ISSUE THREAD

**Issue by sangxxh** [tatuylonen/wiktextract#903]
I'm reading the pre-extracted JSONL data `raw-wiktextract-data.jsonl.gz`.

Some of the words that have more than 1 entry in `inflection_templates` actually contain an obsolete template. But it's not indicated in the JSON.

Example:
See https://en.wiktionary.org/wiki/M%C3%A4rz#German. The second template is indicated as obsolete:

```
====Declension====
{{de-ndecl|m,-[predominant]:es[less common]}}
; Obsolete declension:
{{de-ndecl|m.weak}}
```

In the extracted file, the second template is not indicated as obsolete:

```json
    "inflection_templates": [
        {
            "name": "de-ndecl",
            "args": {
                "1": "f"
            }
        },
        {
            "name": "de-ndecl",
            "args": {
                "1": "f.gen:en.dat:en"
            }
        }
    ],
```

Can a property be added for `inflection_templates` entry to indicate the template as obsolescent?

**Comment by xxyzz** [comment]
IMO "_templates" fields are probably not designed for end users, they only contain template name and parameters. Users should use the "forms" field, and ideally an "obsolete" tag will be added to data in "forms" list.

**Comment by sangxxh** [comment]
> IMO "_templates" fields are probably not designed for end users, they only contain template name and parameters. Users should use the "forms" field, and ideally an "obsolete" tag will be added to data in "forms" list.

Given that it's part of the raw declension template, I think it's handy to have it in the `inflection_templates` data for the sake of data completeness. 
Adding an obsolete flag to the `forms` field as you suggested is essential. 👍

**Comment by kristian-clausal** [comment]
The `_templates` are for raw template arguments, name and expansion only. We need yet another kludge for parsing this particular construction, because editors can't help themselves. It will be on the backburner.

**Comment by kristian-clausal** [comment]
Should be fixed by #905, and I'm reasonably happy with the relatively low footprint of this giant, ugly terrible hack.

1. `;` should not be used for its bolding properties. It breaks things for accessibility readers, because what it is actually part of a *definition list*, not a general formatter or hierarchy indicator.
2. We have a long preceding kludge that if you have two or more templates in an inflection section, any text between them is associated to the preceding template; it has an "after" text.
3. The `;` is parsed as a list because it's a list, why would it be a heading?

The kludge is to ignore all of these best Wikitext practices and add exceptions just for this case. If you have a `;` list between templates, interpret it as a title, and apply it to every template that follows, until it is overwritten by another `;` list.
